### ISSUE TITLE
Bug: Point selection method selects points in the water

### ISSUE THREAD

**Issue by wietskebrouwer** [TUDelftGeodesy/sarxarray#44]
The function 'slcstack.point_selection(threshold, method="amplitude_dispersion",chunks=5000)' selects also points in the water. 
This is strange since these points have a very high amplitude dispersion. 
 
See the image in the appendix for the selected points. The red points are points that are selected in the water 
![Scherm­afbeelding 2023-11-15 om 16 21 41](https://github.com/MotionbyLearning/sarxarray/assets/113337632/0900ff05-c1cc-4ba9-8980-8e34f04bc17a)

This problem is caused since there are points with nans in the slc stack. When we then apply the point selection method, these points are also selected since the algorithm treats these nan values as very small values

**Comment by rogerkuou** [comment]
Thanks @wietskebrouwer for submitting this issue. Indeed it is weird to have points with such low amplitude dispersion on water. I will take a look at this ASAP.

**Comment by rogerkuou** [comment]
Hi @wietskebrouwer,
I made [this example notebook](https://github.com/MotionbyLearning/sarxarray/files/13379131/44_ps_water.zip) (you need up unzip it) to reproduce the problem, but to me it all works well, no points are selected on the water.

However, I do notice one thing which may cause the problem: in the Zarr file you created:  `computed_slcs_s1_dsc_t037.zarr`, there is only `complex` data but no `amplitude`. Maybe the way you add the `amplitude` changed the value of amplitude on those NaN locations, and making those points popping up.

In the attached Notebook, I was using the `_get_amplitude` function to add the `amplitude` as a delayed object. Then if the `complex` is `np.nan` at the location, the `amplitude` stays `np.nan` and won't pass the amplitude dispersion criterion. Because both `np.nan<value` and `np.nan>value` will return `False`.

My take away from this, is maybe we should make the two internal functions `_get_amplitude` and `_get_phase` public, since they are recommended ways to add these two fields. 

Could you please check if your problem is solved by calling `_get_amplitude`, to compute amplitude?

**Comment by wietskebrouwer** [comment]
Hi @rogerkuou,
Thanks for the update. When I run your notebook, I indeed do not get the points selected at the water. I will now try to implement this way of adding the phase and amplitude data to the stack in my own code to check whether it indeed works.

**Comment by wietskebrouwer** [comment]
I think I already see where the problem is. In my original note I send to you, I needed to add the mother to the slc stack. See image:
![Scherm­afbeelding 2023-11-16 om 17 04 01](https://github.com/MotionbyLearning/sarxarray/assets/113337632/0e75b3ed-5a18-4931-a3d9-230d1f0e8a2f)


Probably this is causing the issue. Because on the stack where I manually added the mother image (see my notebook) I still have to problem that points in the water are selected. Also when I apply the _get_amplitude and _get_phase functions you suggested.

**Comment by rogerkuou** [comment]
Hi @wietskebrouwer, I found this is related to the behavior of [xarray.DataArray.std](https://docs.xarray.dev/en/stable/generated/xarray.DataArray.std.html) and [xarray.DataArray.mean](https://docs.xarray.dev/en/stable/generated/xarray.DataArray.mean.html). By default, they ignore the NaN values, but not returning NaN.

For the points on the water, it appears that they have zero amplitude value on reference image, and NaN on others. Then the NAD and NMAD are all 0, which passes the thresholding of "<threshold", hence they are selected.

I made a fix in PR#51. No extra excluding needed since NaN should not pass the thresholding. And I am not infavor of adding a thresholding specifically on NaN since it is redundant and may cause unexpected behavior.

For your implementation, I suggest to change it like this:

```python
def point_selection(self, threshold, method, chunks=1000):
        
        ......
        match method:
            case "nad":
                #mask = self._nad() < threshold
                # mask = np.logical_and(~np.isnan(self._nad()), self._nad() < threshold)
                mask = self._nad() < threshold # This should be enough


            case "nmad":
                #mask = self._nmad() < threshold
                # mask = np.logical_and(~np.isnan(self._nmad()), self._nmad() < threshold)
                mask = self._nmad() < threshold
            case _:
                raise NotImplementedError
      ........

def _nad(self, chunk_azimuth=500, chunk_range=500):
        # Time dimension order
        t_order = list(self._obj.dims.keys()).index("time")

        # Rechunk to make temporal operation more efficient
        amplitude = self._obj.amplitude.chunk(
            {"azimuth": chunk_azimuth, "range": chunk_range, "time": -1}
        )

        nad = amplitude.std(axis=t_order, skipna=False) / (
            amplitude.mean(axis=t_order, skipna=False) + np.finfo(amplitude.dtype).eps
        )  # adding epsilon to avoid zero division

        return nad


    def _nmad(self, chunk_azimuth=500, chunk_range=500):
        """
        Estimate the NMAD for a xarray dataset. 
        """
        # Time dimension order
        t_order = list(self._obj.dims.keys()).index("time")
        
        # Rechunk to make temporal operation more efficient
        amplitude = self._obj.amplitude.chunk(
            {"azimuth": chunk_azimuth, "range": chunk_range, "time": -1}
        )

        median_amplitude = amplitude.median(axis=t_order, skipna=False)
        mad = (np.abs(amplitude - median_amplitude)).median(dim="time", skipna=False)
        nmad = mad / (median_amplitude + np.finfo(amplitude.dtype).eps)

        return nmad
```

Could you please try if this works?
